integration_tests/base_routes.py

```diff
@@ -706,4 +725,13 @@ async def async_exception_post(_: Request):
     app.add_view("/sync/view", SyncView)
     app.add_view("/async/view", AsyncView)
     app.include_router(sub_router)
+
+    class BasicAuthHandler(AuthenticationHandler):
+        def authenticate(self, request: Request) -> Optional[Identity]:
+            token = self.token_getter.get_token(request)
+            if token == "valid":
+                return Identity(claims={"key": "value"})
+            return None
```

Comment:
I think this can be a possible improvement, I just made the overall architecture available with this PR but we could build on it to provide default `TokenGetter` and `AuthenticationHandler`. For example, an `AuthenticationHandler` that decodes a JWT token, and a `TokenGetter` that gets it from the cookies of the request. Right now I just implemented a `TokenGetter` that retrieve the token from the `Authorization` headers. In the future we can decide if we want to split those helper classes in a separate package :slightly_smiling_face:

Reply:
> For example, an AuthenticationHandler that decodes a JWT token, and a TokenGetter that gets it from the cookies of the request. Right now I just implemented a TokenGetter that retrieve the token from the Authorization headers. In the future we can decide if we want to split those helper classes in a separate package 🙂

This sounds great. Can definitely add it in the future 😄